src/fiatExchanges/CoinbasePaymentSection.tsx

```diff
@@ -1,29 +1,64 @@
 import React from 'react'
 
+import { generateOnRampURL } from '@coinbase/cbpay-js'
+import { useAsync } from 'react-async-hook'
 import { useTranslation } from 'react-i18next'
 import { Image, StyleSheet, Text, View } from 'react-native'
 import { useSelector } from 'react-redux'
 import { coinbasePayEnabledSelector } from 'src/app/selectors'
 import Touchable from 'src/components/Touchable'
 import { FetchProvidersOutput } from 'src/fiatExchanges/utils'
+import { readOnceFromFirebase } from 'src/firebase/firebase'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
 import colors from 'src/styles/colors'
 import fontStyles from 'src/styles/fonts'
+import { CiCoCurrency } from 'src/utils/currencies'
+import { walletAddressSelector } from 'src/web3/selectors'
 
 export interface CoinbasePaymentSectionProps {
+  digitalAsset: CiCoCurrency
+  cryptoAmount: number
   coinbaseProvider: FetchProvidersOutput | undefined
 }
 
-export function CoinbasePaymentSection({ coinbaseProvider }: CoinbasePaymentSectionProps) {
+export function CoinbasePaymentSection({
+  digitalAsset,
+  cryptoAmount,
+  coinbaseProvider,
+}: CoinbasePaymentSectionProps) {
   const { t } = useTranslation()
   const coinbasePayEnabled = useSelector(coinbasePayEnabledSelector)
+  const walletAddress = useSelector(walletAddressSelector)!
+  const allowedAssets = [CiCoCurrency.CELO]
```

Comment:
super nit: we check if the `digitalAsset` is one from a list of `allowedAssets`, but when generating the onramp URL, we hardcode it to `CGLD` (corresponding to `CiCoCurrency.CELO`). Can we just similarly hardcode a check that `digitalAsset === CiCoCurrency.CELO`? Once/if Coinbase allows multiple different assets, we'll have to determine which asset to give to the flow by using a map from `digitalAsset`, at which point a list for `allowedAssets` seems more appropriate.

Reply:
Fair enough. It was a half way modular attempt.